crates/proto/src/rr/record_data.rs

```diff
@@ -691,6 +691,9 @@ pub enum RData {
         rdata: NULL,
     },
 
+    /// Update record with RDLENGTH = 0 (RFC2136)
+    Update(RecordType),
```

Comment:
I wonder if we should more explicitly call this a “delete” as that’s what a RDLENGTH of zero means.

Reply:
> that’s what a RDLENGTH of zero means.

not always. the records that go in the prerequisite section can have RDLENGTH=0 and that does not necessarily mean that they themselves are delete operations or that they are included in a message that will perform a delete operation. examples of where `Update` (RDLENGTH=0) appears in non-delete operations in the codebase:

- [append](https://github.com/hickory-dns/hickory-dns/blob/76f717d10f6d6a5ed176c275a416c27b0b9e4ff8/crates/proto/src/op/update_message.rs#L254) - section 2.4.1 of RFC2136
- [create](https://github.com/hickory-dns/hickory-dns/blob/76f717d10f6d6a5ed176c275a416c27b0b9e4ff8/crates/proto/src/op/update_message.rs#L184) - section 2.4.3 of RFC2136

so I would not rename this to `Delete` but perhaps it should be called `Update0` to indicate that RDLENGTH is 0?

Reply:
Ah, good catch. Update0, UpdateEmpty, yeah, any of those make sense to me.

Reply:
amended the commit to use `Update0`